airbyte-integrations/connectors/source-google-drive/source_google_drive/spec.py

```diff
@@ -55,6 +55,8 @@ class Config:
         description="URL for the folder you want to sync. Using individual streams and glob patterns, it's possible to only sync a subset of all files located in the folder.",
         examples=["https://drive.google.com/drive/folders/1Xaz0vXXXX2enKnNYU5qSt9NS70gvMyYn"],
         order=0,
+        pattern="^https://drive.google.com/.+",
+        pattern_descriptor="https://drive.google.com/folders/MY-FOLDER-ID",
```

Comment:
Looks like a small typo here.

```suggestion
        examples=["https://drive.google.com/drive/folders/1Xaz0vXXXX2enKnNYU5qSt9NS70gvMyYn"],
        order=0,
        pattern="^https://drive.google.com/.+",
        pattern_descriptor="https://drive.google.com/drive/folders/MY-FOLDER-ID",
```

Here is an example of a URL I tried and failed:

- `https://drive.google.com/drive/u/0/folders/1J7ARNrRWGD437RyTfA8JzeCM8Oc9EZOU` (a real link, but nothing confidential here and no write permissions.)

In my case, I had to remove `/u/0` from the URL manually, and then it would work.

What do you think of Expanding pattern also, something like this?:

```suggestion
        examples=["https://drive.google.com/drive/folders/1Xaz0vXXXX2enKnNYU5qSt9NS70gvMyYn"],
        order=0,
        pattern="^https://drive.google.com/drive/folders/.+",
        pattern_descriptor="https://drive.google.com/drive/folders/MY-FOLDER-ID",
```

Reply:
Also, I'm not sure if this version works:  `https://drive.google.com/drive/folders/1J7ARNrRWGD437RyTfA8JzeCM8Oc9EZOU?usp=sharing`. 

This is what I get if I go to the sharing screen and click "Copy Link". My vague memory is that it didn't work when I tried it, but it looks like the only difference is the suffix `?usp=sharing` - which we could probably automatically trim from the URL if it is exists.

Wdyt?

Reply:
The util function I wrote can deal with `https://drive.google.com/drive/u/0/folders/1J7ARNrRWGD437RyTfA8JzeCM8Oc9EZOU`and with `https://drive.google.com/drive/folders/1J7ARNrRWGD437RyTfA8JzeCM8Oc9EZOU?usp=sharing`. To make it easy for the user I don't want to block these via pattern (if it doesn't start with `https://drive.google.com`, then it's definitely off).

With the current setup it should work whether you have the u/0 thing in there or not and whether you use the share link with the query param.